drivers/hud_twenty_twenty_to_twenty_twenty_two/app/models/hud_twenty_twenty_to_twenty_twenty_two/enrollment/update_r7_columns.rb

```diff
@@ -0,0 +1,51 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+#
+
+module HudTwentyTwentyToTwentyTwentyTwo::Enrollment
+  class UpdateR7Columns
+    COLUMNS = [
+      'UrgentReferral',
+      'TimeToHousingLoss',
+      'AnnualPercentAMI',
+      'LiteralHomelessHistory',
+      'ClientLeaseholder',
+      'HOHLeasesholder',
+      'SubsidyAtRisk',
+      'EvictionHistory',
+      'CriminalRecord',
+      'IncarceratedAdult',
+      'PrisonDischarge',
+      'SexOffender',
+      'DisabledHoH',
+      'CurrentPregnant',
+      'SingleParent',
+      'DependentUnder6',
+      'HH5Plus',
+      'CoCPrioritized',
+      'HPScreeningScore',
+      'ThresholdScore',
+    ].freeze
+
+    # Pending HUD guidance, this preserves existing columns, and sets the new ones to nil
+    # TargetScreenReqd is set if thee are any values in the columns
+    def process(row)
+      COLUMNS.each do |column|
+        row[column] = nil unless row.keys.include?(column)
```

Comment:
This seems like it could be really slow since it has to pull keys off the row for _every_ row.  won't all of the rows in a given file have the same keys?  also, we should probably use `has_key?` instead of `keys.include?`  I believe `keys` makes an array of the hash keys, then checks to see if the item is in the array, `has_key?` seems to have some other magic going on: https://apidock.com/ruby/Hash/has_key%3f.

Reply:
maybe some version of `row[column] ||= nil`?